core/src/main/java/org/frankframework/jdbc/MessageStoreListener.java

```diff
@@ -122,28 +132,75 @@ public void configure() throws ConfigurationException {
 	}
 
 	public void extractSessionKeyList() {
-		if (sessionKeys != null) {
-			sessionKeysList = StringUtil.split(sessionKeys);
+		sessionKeysList = StringUtil.split(sessionKeys);
+	}
+
+	@Override
+	protected RawMessageWrapper<Serializable> extractRawMessage(ResultSet rs) throws JdbcException {
+		try (InputStream blobStream = JdbcUtil.getBlobInputStream(getDbmsSupport(), rs, getMessageField(), isBlobsCompressed());
+		 ObjectInputStream ois = new RenamingObjectInputStream(blobStream)) {
+			String key = getStringFieldOrNull(rs, getKeyField());
+			String cid = getStringFieldOrNull(rs, getCorrelationIdField());
+			String mid = getStringFieldOrNull(rs, getMessageIdField());
+			Object rawMessage = ois.readObject();
+
+			if (rawMessage instanceof MessageWrapper<?>) {
+				//noinspection unchecked
+				return (MessageWrapper<Serializable>) rawMessage;
+			}
+
+			RawMessageWrapper<Serializable> rawMessageWrapper = new RawMessageWrapper<>((Serializable)rawMessage, mid != null ? mid : key, cid);
+			if (key != null) {
+				rawMessageWrapper.getContext().put(PipeLineSession.STORAGE_ID_KEY, key);
+			}
+			return rawMessageWrapper;
+		} catch (Exception e) {
+			throw new JdbcException(e);
 		}
+
+	}
+
+	private String getStringFieldOrNull(ResultSet rs, String columnLabel) throws SQLException {
```

Comment:
Je gooit hier een sql exception (van .getString) maar je vangt hem voor .findColumn wel af. Wil je hier niet beide in de try of juist beide erbuiten?

Reply:
ik denk dat het meer iets is in de zin van, get-value-if-exists or null. en als de get-value een exceptie gooit dan is er iets fout.

Reply:
Inderdaad, dat is het. Ik weet niet of die column in de resultset zit maar de manier om daarachter te komen is `rs.findColumn()`. En die geeft niet iets als `-1` terug zoals `String.indexOf()` maar die gooit een exception.